code/modules/research/xenobiology/crossbreeding/chilling.dm

```diff
@@ -0,0 +1,311 @@
+/*
+Chilling extracts:
+	Have a unique, primarily defensive effect when
+	filled with 10u plasma and activated in-hand.
+*/
+/obj/item/slimecross/chilling
+	name = "chilling extract"
+	desc = "It's cold to the touch, as if frozen solid."
+	effect = "chilling"
+	container_type = INJECTABLE | DRAWABLE
+	icon_state = "chilling"
+
+/obj/item/slimecross/chilling/Initialize()
+	. = ..()
+	create_reagents(10)
+
+/obj/item/slimecross/chilling/attack_self(mob/user)
+	if(!reagents.has_reagent("plasma",10))
+		to_chat(user, "<span class='warning'>This extract needs to be full of plasma to activate!</span>")
+		return
+	reagents.remove_reagent("plasma",10)
+	to_chat(user, "<span class='notice'>You squeeze the extract, and it absorbs the plasma!</span>")
+	playsound(src, 'sound/effects/bubbles.ogg', 50, 1)
+	playsound(src, 'sound/effects/glassbr1.ogg', 50, 1)
+	do_effect(user)
+
+/obj/item/slimecross/chilling/proc/do_effect(mob/user) //If, for whatever reason, you don't want to delete the extract, don't do ..()
+	qdel(src)
+	return
+
+/obj/item/slimecross/chilling/grey
+	colour = "grey"
+
+/obj/item/slimecross/chilling/grey/do_effect(mob/user)
+	user.visible_message("<span class='notice'>[src] produces a few small, grey cubes</span>")
+	for(var/i in 1 to 3)
+		new /obj/item/barriercube(get_turf(user))
+	..()
+
+/obj/item/slimecross/chilling/orange
+	colour = "orange"
+
+/obj/item/slimecross/chilling/orange/do_effect(mob/user)
+	user.visible_message("<span class='danger'>[src] shatters, and lets out a jet of heat!</span>")
+	for(var/turf/T in orange(get_turf(user),2))
+		if(get_dist(get_turf(user), T) > 1)
+			new /obj/effect/hotspot(T)
```

Comment:
You still need to deal with the issue of hotspots getting created on the other side of walls.

Reply:
Er, not to question too much, but why would that be a *bad* thing...? Phlogiston and such does that.

Reply:
A lot of ss13's code is terrible for various reasons, doesn't mean you have to follow suit.

Reply:
Probably the simplest and most interesting way to solve the issue would be to make it shoot actual fire projectiles

Reply:
Ooh, good plan!